sector-builder-ffi/src/api.rs

```diff
@@ -542,6 +546,117 @@ pub unsafe extern "C" fn sector_builder_ffi_read_piece_from_sealed_sector(
     })
 }
 
+/// Imports a sector sealed elsewhere and takes ownership of its metadata,
+/// cache directory, and sealed sector file.
+///
+#[no_mangle]
+pub unsafe extern "C" fn sector_builder_ffi_import_sealed_sector(
+    ptr: *mut SectorBuilder,
+    sector_id: u64,
+    sector_cache_dir_path: *const libc::c_char,
+    sealed_sector_path: *const libc::c_char,
+    seal_ticket: FFISealTicket,
+    seal_seed: FFISealSeed,
+    comm_r: &[u8; 32],
+    comm_d: &[u8; 32],
+    p_aux_comm_c: &[u8; 32],
+    p_aux_comm_r_last: &[u8; 32],
+    proof_ptr: *mut u8,
+    proof_len: libc::size_t,
+    pieces_ptr: *mut FFIPieceMetadata,
+    pieces_len: libc::size_t,
+) -> *mut types::ImportSealedSectorResponse {
+    catch_panic_response(|| {
+        init_log();
+
+        info!("import_sealed_sector: {}", "start");
+
+        let mut response: types::ImportSealedSectorResponse = Default::default();
+
+        let comm_r_last = PedersenDomain::try_from_bytes(p_aux_comm_r_last);
+        let comm_c = PedersenDomain::try_from_bytes(p_aux_comm_c);
+
+        if comm_r_last.is_err() {
+            response.status_code = FCPResponseStatus::FCPUnclassifiedError;
+            response.error_msg = rust_str_to_c_str("cannot xform comm_r_last to PedersenDomain");
+            info!("seal_commit: finish");
+            return raw_ptr(response);
+        }
+
+        if comm_c.is_err() {
+            response.status_code = FCPResponseStatus::FCPUnclassifiedError;
+            response.error_msg = rust_str_to_c_str("cannot xform comm_c to PedersenDomain");
+            info!("seal_commit: finish");
+            return raw_ptr(response);
+        }
+
+        let pieces: Vec<PieceMetadata> = Vec::from_raw_parts(pieces_ptr, pieces_len, pieces_len)
+            .iter()
+            .map(|meta| PieceMetadata {
+                piece_key: c_str_to_rust_str(meta.piece_key).to_string(),
+                num_bytes: UnpaddedBytesAmount(meta.num_bytes),
+                comm_p: meta.comm_p,
+            })
+            .collect();
+
+        match (*ptr).import_sealed_sector(
+            SectorId::from(sector_id),
+            c_str_to_pbuf(sector_cache_dir_path),
+            c_str_to_pbuf(sealed_sector_path),
+            seal_ticket.into(),
+            seal_seed.into(),
+            *comm_r,
+            *comm_d,
+            PersistentAux {
+                comm_c: comm_c.unwrap(),
+                comm_r_last: comm_r_last.unwrap(),
+            },
+            pieces,
+            Vec::from_raw_parts(proof_ptr, proof_len, proof_len).to_vec(),
```

Comment:
should be `slice` not `Vec` as above I believe

Reply:
addressed as per your comments